src/GraphQL/Schema.hs

```diff
@@ -119,26 +121,11 @@ newtype TypeExtensionDefinition = TypeExtensionDefinition ObjectTypeDefinition
 data InputType = DefinedInputType InputTypeDefinition | BuiltinInputType Builtin deriving (Eq, Show)
 
 
-data InputTypeDefinition =
-  -- tom: Next line is wrong I think? We should use InputObjectTypeDefinition?
-  InputTypeDefinitionObject        ObjectTypeDefinition
+data InputTypeDefinition
+  = InputTypeDefinitionObject        InputObjectTypeDefinition
   | InputTypeDefinitionScalar        ScalarTypeDefinition
   | InputTypeDefinitionEnum          EnumTypeDefinition
-  | InputTypeDefinitionInputObject   InputObjectTypeDefinition
   deriving (Eq, Show)
 
 
-type DefaultValue = InputValue
-
-data InputValue
-  = ValueInt Int
-  | ValueFloat Double
-  | ValueBoolean Bool
-  | ValueString Text
-  | ValueEnum Name
-  | ValueList ListValue
-  | ValueObject [ObjectField] deriving (Eq, Show)
-
-data ListValue = ListValue [InputValue] deriving (Eq, Show)
-data ObjectValue = ObjectValue [ObjectField] deriving (Eq, Show)
-data ObjectField = ObjectField Name InputValue deriving (Eq, Show)
+type DefaultValue = Value
```

Comment:
What's the use case for `DefaultValue`? Can you add a comment (doesn't need to be a full doc comment, just a note what it's purpose will be)